src/protocol/context/semi_honest.rs

```diff
@@ -0,0 +1,80 @@
+use crate::ff::Field;
+use crate::helpers::messaging::{Gateway, Mesh};
+use crate::helpers::Role;
+use crate::protocol::context::{ContextInner, MaliciousProtocolContext, ProtocolContext};
+use crate::protocol::malicious::SecurityValidatorAccumulator;
+use crate::protocol::prss::{
+    Endpoint as PrssEndpoint, IndexedSharedRandomness, SequentialSharedRandomness,
+};
+use crate::protocol::{Step, Substep};
+use crate::secret_sharing::Replicated;
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+/// Context for protocol executions suitable for semi-honest security model, i.e. secure against
+/// honest-but-curious adversary parties.
+#[derive(Clone, Debug)]
+pub struct SemiHonestProtocolContext<'a, F: Field> {
+    inner: ContextInner<'a>,
+    _marker: PhantomData<F>,
```

Comment:
As noted, I think we can remove `F` from the type and have it attach to `Self::Share`.